File: Session11/Java/LevelOrderTraversal.java
https://leetcode.com/problems/binary-tree-level-order-traversal/

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        List<List<Integer>> result = new ArrayList<List<Integer>>();
       
        if(root == null) {
            return result;
        }
       
       Queue<TreeNode> queue = new LinkedList<TreeNode>();
       queue.add(root);
       
        while(!queue.isEmpty()) {
            List<Integer> level = new ArrayList<Integer>(); 
            int n = queue.size();
            for(int i=0;i<n;i++) {
                TreeNode node = queue.poll();
                level.add(node.val);
                if(node.left!=null) {
                    queue.add(node.left);
                }
                if(node.right!=null) {
                    queue.add(node.right);
                }
            }
            result.add(level);
        }
        
        return result;
    }
}

File: Session11/C++/LevelOrderTraversal.cpp
https://leetcode.com/problems/binary-tree-level-order-traversal/

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        if(root==NULL) {
            return {};
        }
        
        queue<TreeNode*> q;
        q.push(root);
        vector<vector<int>> res;
        
        while(!q.empty()) {
            vector<int> level;
            int n=q.size(); 
            for(int i=0; i<n; i++) {
                TreeNode* node=q.front();
                q.pop();
                level.push_back(node->val);
                if(node->left) {
                    q.push(node->left);
                }
                if(node->right) {
                    q.push(node->right);
                }       
            }
            res.push_back(level);
        }
        
        return res;
    }
};